### DISCUSSION TITLE
How to adjust height of a cylinder

### DISCUSSION THREAD

**Discussion by Sam777Sam** [idaholab/moose#21715] [Q&A Meshing] [answered]
Hi,

I am using the "file=cyl-tet.e" as my mesh. However the height of the cylinder is given as 5 cm and I would like to make it 15 cm. How can I do that? also is there a link would elaborate more about mesh adjustments? Your help is appreciated. Thank you!

Regards,
Sam

**Comment by GiudGiud** [comment, marked as answer]
Hello

there s a few ways you could do this. You could load the file several times and stitch it top to bottom

you could generate a 2D circle then extrude it. 

Please look at our mesh generators to find the basic operations you can use to generate cylinders 

Guillaume

**Reply by Sam777Sam** [reply]
Thank you Guillaume!

1- I'm using ParaView program to open the "file=cyl-tet.e" and I couldn't  stich it top to bottom. How can I do that? and do you suggest me to use another program to do that (what is the best program for editing the shapes)?
2- Also, I did the 2D circle which is same as the rectangular one. I was following a 2D circle example in Moose and here is my code for the mesh:
 [Mesh]
type = GeneratedMesh
xmin = 0.0
ymin = 0.0
xmax = 0.5
ymax = 15
nx = 5
ny = 150
[]
Is this correct? and does the Moose understand that this is a circle not a rectangle?

3- I looked at the mesh generators and unfortunately things doesn't seem clear to me, is there a good source or videos talking about mesh adjustments and what kind of functions that can be added in order to edit the meshes. 

I'm new to Moose and your help is highly appreciated. Thank you for your time!

Best,
Sam

**Reply by GiudGiud** [reply]
Hello

1) To stitch you may use MOOSE mesh generators. I dont know how to do this with paraview. 

2) This is not a circle mesh, this is a rectangle mesh. I recommend you look more at the mesh generator system:
https://mooseframework.inl.gov/syntax/Mesh/index.html#20d24524-c90b-462f-b4da-e1208cc17dcb

you may generate a circle with this one:
https://mooseframework.inl.gov/source/meshgenerators/ConcentricCircleMeshGenerator.html

You should try to run some of the inputs in moose/test/tests/mesh_generators to get familiar with that system. Go in each folder, run the input and look at the output exodus to see what the mesh generators (nested in the [Mesh] block) did

Guillaume
